.github/workflows/my-ci-pipeline.yml

```
@@ -11,6 +11,25 @@ jobs:
     name: Build and test
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v2
-    - run: docker build --target test --tag todo-app:test .
-    - run: docker run todo-app:test
\ No newline at end of file
+    - uses: actions/checkout@v3
+    - uses: satackey/action-docker-layer-caching@v0.0.11
+      continue-on-error: true
+    - run: docker build --target test --tag todo-app .
+    - run: docker run todo-app
+  deploy:
+    name: push image to dockerhub
+    needs: build
+    runs-on: ubuntu-latest
+    if: github.ref_name == 'module9' && github.event_name == 'push'
+    steps:
+      - name: Login to Dockerhub
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      - uses: actions/checkout@v3
+      - uses: satackey/action-docker-layer-caching@v0.0.11
+        continue-on-error: true
+      - run: docker build --target production --tag "amtronics/todo-app:$GITHUB_SHA" .
+      - run: docker push "amtronics/todo-app:$GITHUB_SHA"
```

This will work for pushing your latest image, but unless you're manually updating the tag every time Azure isn't going to pull your latest image because it doesn't know which SHA to use. Typically we'd solve that by pushing a `latest` tag or similar, in addition to specific versions if we want them, so e.g. potentially like:
```suggestion
      - run: docker build --target production --tag "amtronics/todo-app:$GITHUB_SHA" --tag "amtronics/todo-app:prod" .
      - run: docker push "amtronics/todo-app:$GITHUB_SHA" --all-tags
```
(I haven't tested that `--all-tags` argument so do check it does what you want, otherwise we could just push the `:prod` image in a separate step, and DockerHub should recognise that it's just a new tag for an existing image)